### PR TITLE
treewide: fix string escapes

### PR DESCRIPTION
**Pull request by volth**
Nix parser is very relaxing and allows any character after `\` in strings
When the character following `\` is not one of the six which require escaping (`"` `r` `n` `$` `\` `t`) then `\` is silently skipped

so `"\1"` is the same as just`"1"`

This makes writing string constants which have `\` character (regular expressions, etc) challenging and error-prone.

### CODE REVIEW

File: pkgs/development/haskell-modules/hackage-packages.nix [hunk 1]
```diff
@@ -57630,7 +57630,7 @@ self: {
        version = "0.1.0.1";
        sha256 = "0rwi5zwvqn18g7qyp9aw51w3yzkqbff9g7rcqdk1l871fvq8qhha";
        libraryHaskellDepends = [ base ];
-       description = "Haskell operator `g ... f = \x1 .. xn -> g (f x1 .. xn)`.";
+       description = "Haskell operator `g ... f = \\x1 .. xn -> g (f x1 .. xn)`.";
```
Comment (volth) [inline pkgs/development/haskell-modules/hackage-packages.nix]: @peti please fix `haskell-package.nix` generator

**Comment by edolstra** [comment]
If this is used in Nixpkgs, then we *can't* change the parser for reasons of compatibility.

**Comment by volth** [comment]
yes, than better just print a warning

**Comment by volth** [comment]
> If this is used in Nixpkgs, then we _can't_ change the parser for reasons of compatibility.

At least `\u` was never user, so it could be reused for `\uXXXX` :)

**Comment by volth** [comment]
@edolstra can you move the discussion above to more relevant https://github.com/NixOS/nix/pull/3093 ?
This PR is not about changing the parser, it is about fixing existing typos, some of them are security-related  (`\.php` -> `\\.php`).

**Pull request merged**